elasticdl/python/master/pod_manager.py

```diff
@@ -230,30 +306,19 @@ def start(self):
     def add_pod_event_callback(self, pod_event_callback):
         self._pod_event_callbacks.append(pod_event_callback)
 
-    def _init_job_pod_status(self):
-        # worker id to (pod name, ip, phase) mapping
-        # phase: None/Pending/Running/Succeeded/Failed/Unknown
-        #   None: worker was just launched, haven't received event yet.
-        #   Pending: worker pod not started yet
-        #   Running: worker pod is running
-        #   Succeeded: worker pod finishes all tasks and terminates with
-        #       no issue.
-        #   Failed: worker pod is killed for some reason
-        #   Unknown: unknown
-        self._worker_pods_ip_phase = {}
-        # pod name to worker id mapping
-        self._worker_pod_name_to_id = {}
-        # worker ids whose pods are not created
-        self._not_created_worker_id = []
-
-        self._relaunch_deleted_live_worker = True
+    def _init_pod_status(self):
+        # _pod_info_cache is a dict. The key is the PodType. The value
+        # is also a dict. The value is a dict mapping pod_name to PodInfo
+        # object.
+        self._pod_info_cache = {PodType.PS: {}, PodType.WORKER: {}}
 
-        self._ps_pods_phase = {}
-        self._ps_pod_name_to_id = {}
-        self._relaunch_deleted_live_ps = True
+        # worker ids for the pods which are not created.
+        # We will try multiple times in the background to create the pod
+        # using the id in the list until success.
+        self._not_created_worker_id = []
 
-        self._failed_pods = []
-        self.all_workers_exited = False
+        self._relaunch_worker = True
+        self._relaunch_ps = True
```

Comment:
We never relaunch ps, so `self._relaunch_ps` is not needed.

Reply:
Good idea, removed.